compiler/src/iree/compiler/GlobalOptimization/SetEncoding.cpp

```diff
@@ -103,28 +105,53 @@ makeEncoding(OpBuilder &builder, IREE::LinalgExt::EncodingUser user,
       context, userAttr, roleAttr, operandElemTypesAttr, originalTypeAttr);
 }
 
+static Value castEncodedResult(OpBuilder &builder, Location loc, Value encoded,
+                               Value castResult,
```

Comment:
If `std::optional<CastOpInterface> srcCastOp` is passed to the function, you don't need to call `getDefiningCastOp` again.